app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt

```diff
@@ -376,3 +424,209 @@ fun InputChipFun(
+                            pickPhotoActivity.launch(
+                                PickVisualMediaRequest(
+                                    ActivityResultContracts.PickVisualMedia.ImageOnly
+                                )
+                            )
+                        }
+                        .testTag("profile-picture-image"),
+                painter =
+                    if (picture != null) {
+                        BitmapPainter(picture!!.asImageBitmap())
+                    } else {
+                        painterResource(R.drawable.echologoround)
+                    },
+                contentDescription = "",
+                alpha = 0.5f
+            )
+            Icon(
+                imageVector = Icons.Outlined.Edit,
+                contentDescription = stringResource(R.string.profile_creation_edit_picture),
+                modifier = Modifier.align(Alignment.Center)
+            )
+        }
+        IconButton(
+            modifier =
+                Modifier.align(Alignment.End)
+                    .offset(x = 10.dp, y = (-10).dp)
+                    .testTag("profile-picture-delete"),
+            onClick = { onPictureChange(null) }
+        ) {
+            Icon(
+                imageVector = Icons.Outlined.Delete,
+                contentDescription = stringResource(R.string.profile_creation_delete_picture),
+            )
+        }
+    }
+
+    if (showPictureDialog) {
+        Dialog(onDismissRequest = { showPictureDialog = false }) {
+            PictureTransformer(
+                picture = rawPicture!!,
+                onCancel = { showPictureDialog = false },
+                onConfirm = { newPicture ->
+                    showPictureDialog = false
+                    onPictureChange(newPicture)
+                }
+            )
+        }
+    }
+}
+
+/**
+ * Picture edition dialog which allow the user to center the picture
+ *
+ * @param picture: the picture to center
+ * @param onConfirm: callback called when the picture is centered
+ * @param onCancel: callback when the action is canceled
+ */
+@Composable
+fun PictureTransformer(
+    picture: Bitmap,
+    onConfirm: (picture: Bitmap) -> Unit,
+    onCancel: () -> Unit
+) {
+    val circleRadius = min(picture.width, picture.height) * 3 / 8
+    val minMaxScale = Pair(0.75f, 2f)
+    val maxResolution = 500 // the picture is cropped to output a square
+    var scale by remember { mutableStateOf(1f) }
+    var offset by remember { mutableStateOf(Offset.Zero) }
+
+    // helper function to find out if the new transformation is inside the picture depending on
+    // offset and scale
+    fun isTransformInside(offs: Offset, sca: Float): Boolean {
+        return offs.x > -picture.width * sca / 2f + circleRadius &&
+            offs.y > -picture.height * sca / 2f + circleRadius &&
+            offs.x < picture.width * sca / 2f - circleRadius &&
+            offs.y < picture.height * sca / 2f - circleRadius &&
```

Comment:
you could modularize the return statement by defining each set of conditions as variables and then using them in the return statement. This would also allow someone in the future to easily edit the conditions without much hassle.